testground/benchmark/benchmark/stateless.py

```diff
@@ -38,149 +38,193 @@
-        (outdir / VALIDATOR_GROUP).mkdir(parents=True, exist_ok=True)
-        (outdir / FULLNODE_GROUP).mkdir(parents=True, exist_ok=True)
-
-        peers = []
-        for i in range(validators):
-            print("init validator", i)
-            ip = hostname_template.format(index=i)
-            peers.append(init_node_local(cli, outdir, VALIDATOR_GROUP, i, ip))
-        for i in range(fullnodes):
-            print("init fullnode", i)
-            ip = hostname_template.format(index=i + validators)
-            peers.append(init_node_local(cli, outdir, FULLNODE_GROUP, i, ip))
-
-        print("prepare genesis")
-        # use a full node directory to prepare the genesis file
-        genesis = gen_genesis(cli, outdir / FULLNODE_GROUP / "0", peers, genesis_patch)
-
-        print("patch genesis")
-        # write genesis file and patch config files
-        for i in range(validators):
-            patch_configs_local(
-                peers, genesis, outdir, VALIDATOR_GROUP, i, config_patch, app_patch
-            )
-        for i in range(fullnodes):
-            patch_configs_local(
-                peers,
-                genesis,
-                outdir,
-                FULLNODE_GROUP,
-                i,
-                config_patch,
-                app_patch,
-            )
-
-        print("write config")
-        cfg = {
-            "validators": validators,
-            "fullnodes": fullnodes,
-            "num_accounts": num_accounts,
-            "num_txs": num_txs,
-            "validator-generate-load": options.get("validator-generate-load", True),
-        }
-        (outdir / "config.json").write_text(json.dumps(cfg))
-
-    def patchimage(
-        self,
-        toimage,
-        src,
-        dst="/data",
-        fromimage="ghcr.io/crypto-org-chain/cronos-testground:latest",
-    ):
-        """
-        combine data directory with an exiting image to produce a new image
-        """
-        with tempfile.TemporaryDirectory() as tmpdir:
-            tmpdir = Path(tmpdir)
-            shutil.copytree(src, tmpdir / "out")
-            content = f"""FROM {fromimage}
+@click.group()
+def cli():
+    pass
+
+
+def validate_json(ctx, param, value):
+    try:
+        return json.loads(value)
+    except json.JSONDecodeError:
+        raise click.BadParameter("must be a valid JSON string")
+
+
+@cli.command()
+@click.argument("outdir")
+@click.option("--hostname-template", default=HOSTNAME_TEMPLATE)
+@click.option("--validators", default=3)
+@click.option("--fullnodes", default=7)
+@click.option("--num-accounts", default=10)
+@click.option("--num-txs", default=1000)
+@click.option("--config-patch", default="{}", callback=validate_json)
+@click.option("--app-patch", default="{}", callback=validate_json)
+@click.option("--genesis-patch", default="{}", callback=validate_json)
+@click.option("--validator-generate-load/--no-validator-generate-load", default=True)
+def gen(**kwargs):
```

Comment:
seems can't wrap whole options as before?

Reply:
the `generic-gen` command do that.

Reply:
and it also covers all the options including `outdir` and `hostname-template`.

Reply:
yea, might need modify readme later
```
nix run .#stateless-testcase generic-gen '{
  "outdir": "/tmp/outputs",
  "validators": 1,
  "fullnodes": 1,
  "num_accounts": 2,
  "num_txs": 1,
  "app_patch": {
    "mempool.max-txs": -1
  },
  "config_patch": {
    "mempool.size": 30000
  }
}'
```